### ISSUE TITLE
PT2 Qat eval

### ISSUE THREAD

**Issue by xiguadong** [pytorch/ao#2980]
### 🐛 Describe the bug

In QAT (Quantization-Aware Training), it is necessary to call torch.ao.quantization.allow_exported_model_train_eval(model) along with model.eval() and model.train().

This actually triggers patterns in torch/ao/quantization/pt2e/export_utils.py to replace dropout and batchnorm:
```
    def dropout_train(x):
            return F.dropout(x, p=0.5, training=True, inplace=inplace)

    def dropout_eval(x):
            return F.dropout(x, p=0.5, training=False, inplace=inplace)
    def bn_train(
        x: torch.Tensor,
        bn_weight: torch.Tensor,
        bn_bias: torch.Tensor,
        bn_running_mean: torch.Tensor,
        bn_running_var: torch.Tensor,
    ):
        return F.batch_norm(
            x, bn_running_mean, bn_running_var, bn_weight, bn_bias, training=True
        )

    def bn_eval(
        x: torch.Tensor,
        bn_weight: torch.Tensor,
        bn_bias: torch.Tensor,
        bn_running_mean: torch.Tensor,
        bn_running_var: torch.Tensor,
    ):
        return F.batch_norm(
            x, bn_running_mean, bn_running_var, bn_weight, bn_bias, training=False
        )
```
The above patterns will forcibly modify the p parameter of dropout and the momentum and eps(defaults 1e-5) parameters of batch_norm.

These parameter modifications directly affect the original model's training hyperparameters. I tried do PT2 QAT with YOLOv5, it was found that changes to eps (yolov5 configs 1e-3)  can cause a drop of one point in floating-point precision accuracy.

it confused me by several weeks. 

### Versions

torch2.6

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

**Comment by xiguadong** [comment]
in /torch/ao/quantization/pt2e/qat_utils.py , also force set eps to 1e-5

```
def _get_folded_quantized_qat_conv_bn_pattern(
    is_per_channel: bool,
    has_bias: bool,
    bias_is_quantized: bool,
    conv_fn: Callable,
    bn_is_training: bool,
) -> Callable:
    """
    Quantized QAT conv - bn pattern with bn weights being folded into conv.
    """
    # TODO: allow setting eps
    bn_eps = 1e-5

    def _folded_quantized_qat_conv_bn_pattern(
        x: torch.Tensor,
        conv_weight: torch.Tensor,
        bn_weight: torch.Tensor,
        bn_bias: torch.Tensor,
        bn_running_mean: torch.Tensor,
        bn_running_var: torch.Tensor,
        **kwargs,
    ) -> torch.Tensor:
        conv_weight = _append_qdq(
            conv_weight,
            is_per_channel,
            is_bias=False,
            kwargs=kwargs,
        )
        if has_bias:
            bias = kwargs["conv_bias"]
            if bias_is_quantized:
                bias = _append_qdq(
                    bias,
                    is_per_channel,
                    is_bias=True,
                    kwargs=kwargs,
                )
        else:
            bias = None
        x = conv_fn(x, conv_weight, bias)
        x = F.batch_norm(
            x,
            bn_running_mean,
            bn_running_var,
            bn_weight,
            bn_bias,
            training=bn_is_training,
            eps=bn_eps,
        )
        return x

    return _WrapperModule(_folded_quantized_qat_conv_bn_pattern)
```

**Comment by jerryzh168** [comment]
Thanks for reporting the issue, sorry about this, I think we should have fixed this earlier
also we have moved this flow to torchao: https://github.com/pytorch/ao, and won't be making updates in pytorch
I'll transfer issue to torchao as well

cc @andrewor14 can you take a look?
